lib/fluent/plugin/out_sns.rb

```diff
@@ -32,16 +36,18 @@ def configure(conf)
     def start
       super
       options = {}
-      options[:sns_endpoint] = @sns_endpoint
-      options[:proxy_uri] = @proxy
+      options[:region] = @sns_region
+      options[:http_proxy] = @proxy
       if @aws_key_id && @aws_sec_key
-        options[:access_key_id] = @aws_key_id
-        options[:secret_access_key] = @aws_sec_key
+        options[:credentials] = Aws::Credentials.new(@aws_key_id, @aws_sec_key)
       end
-      AWS.config(options)
+      Aws.config.update(options)
 
-      @sns = AWS::SNS.new
-      @topic = @sns.topics.find{|topic| @sns_topic_name == topic.name}
+      @sns = Aws::SNS::Resource.new
+      @topic = @sns.topics.find{|topic| @sns_topic_name == topic.arn.split(":")[-1]}
```

Comment:
`topic.name` is not available anymore, perhaps we should replace `sns_topic_name` with `sns_topic_arn`?